### PR TITLE
Multi-arch github-action workflow unification

### PR DESCRIPTION
**Pull request by cevich**
This is a port from the podman repository, of substantially the same
workflow with a number of bugfixes and readability improvements
compared to the original.  Same for the README.md updates.

The significant changes compared to the prior implementation are:

* Run periodically instead of only after every master push.
* Add a build for the "testing" image flavor.
* Fix a blank `org.opencontainers.image.source` value.
* Instead of pushing a `main` (or `master`) tagged image, use `latest`.
* Simplify use of env. vars. and workflow vars.

Note: Aside from a `s/podman/buildah/g` this commit makes the
buildah and podman workflows identical.  This is needed to better
support a smooth transition to a future/intended unification effort.

In other words, I intend to develop a single, shared workflow/script
that can be used for all three: skopeo, buildah, and podman.

Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

**Comment by edsantiago** [comment]
```diff
diff --git a/tests/validate/pr-should-include-tests b/tests/validate/pr-should-include-tests
index cd95cf67..4cc843df 100755
--- a/tests/validate/pr-should-include-tests
+++ b/tests/validate/pr-should-include-tests
@@ -49,6 +49,7 @@ filtered_changes=$(git diff --name-status $base $head |
                        fgrep -vx go.mod               |
                        fgrep -vx go.sum               |
                        egrep -v  '^[^/]+\.md$'        |
+                       egrep -v  '^\.github/'         |
                        egrep -v  '^contrib/'          |
                        egrep -v  '^docs/'             |
                        egrep -v  '^hack/'             |
```

**Comment by edsantiago** [comment]
Verified by running this against the podman version and running diff:
```
$ sed -e 's/podman/buildah/g' -e 's/PODMAN/BUILDAH/g' -e 's/Podman/Buildah/g' ...
```
LGTM so far

**Comment by TomSweeneyRedHat** [comment]
@cevich needs a `[NO TESTS NEEDED]` tag to get past the smoke test.

File: contrib/buildahimage/README.md [hunk 1]
```diff
@@ -16,13 +16,24 @@ default to `/`.
...
-  * quay.io/buildah/testing - This image is built using the latest version of Buildah that is or was in updates testing for Fedora.  At times this may be the same as the stable image.  This container image will primarily be used by the development teams for verification testing when a new package is created.  Built with buildahimage/testing/Dockerfile.
-  * quay.io/buildah/stable:version - This image is built 'by hand' using a Fedora based container.  An RPM is first pulled from the [Fedora Updates System](https://bodhi.fedoraproject.org/) and the image is built from there.  For more details, see the Containerfile used to build it, buildahimage/stablebyhand/Containerfile.buildahstable
+  * `quay.io/containers/buildah:<version>` and `quay.io/buildah/stable:<version>` -
+    These images are built when a new buildah version becomes available in
+    Fedora.  These images are intended to be unchanging and stable, they will
+    never be updated by automation once they've been pushed.  For build
+    details, see the configuration used to build it,
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:23]:
```suggestion
    details, see the [buildahimage/stable/Dockerfile](stable/Dockerfile) used to build it.
```

File: contrib/buildahimage/README.md [hunk 2]
```diff
@@ -16,13 +16,24 @@ default to `/`.
...
+    [buildahimage/stable/Dockerfile](stable/Dockerfile).
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:24]:
```suggestion

```

File: contrib/buildahimage/README.md [hunk 3]
```diff
@@ -16,13 +16,24 @@ default to `/`.
...
+  * `quay.io/containers/buildah:latest` and `quay.io/buildah/stable:latest` -
+    Built daily using the same Containerfile as above.  The buildah version
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:26]:
```suggestion
    Built daily using the same Dockerfile as above.  The Buildah version
```

File: contrib/buildahimage/README.md [hunk 4]
```diff
@@ -16,13 +16,24 @@ default to `/`.
...
+    will remain the "latest" available in Fedora, however the other image
+    contents may vary compared to the version-tagged images.
+  * `quay.io/buildah/testing:latest` - This image is built daily, using the
+    latest version of buildah that was in the Fedora `updates-testing` repository.
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:30]:
```suggestion
    latest version of Buildah that was in the Fedora `updates-testing` repository.
```

File: contrib/buildahimage/README.md [hunk 5]
```diff
@@ -16,13 +16,24 @@ default to `/`.
...
-  * quay.io/buildah/upstream - This image is built using the latest code found in this GitHub repository.  When someone creates a commit and pushes it, the image is created.  Due to that the image changes frequently and is not guaranteed to be stable.  Built with buildahimage/upstream/Dockerfile.
-  * quay.io/buildah/testing - This image is built using the latest version of Buildah that is or was in updates testing for Fedora.  At times this may be the same as the stable image.  This container image will primarily be used by the development teams for verification testing when a new package is created.  Built with buildahimage/testing/Dockerfile.
-  * quay.io/buildah/stable:version - This image is built 'by hand' using a Fedora based container.  An RPM is first pulled from the [Fedora Updates System](https://bodhi.fedoraproject.org/) and the image is built from there.  For more details, see the Containerfile used to build it, buildahimage/stablebyhand/Containerfile.buildahstable
+  * `quay.io/containers/buildah:<version>` and `quay.io/buildah/stable:<version>` -
+    These images are built when a new buildah version becomes available in
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:20]:
```suggestion
    These images are built when a new Buildah version becomes available in
```

**Comment by TomSweeneyRedHat** [comment]
A few doc nits, and need to figure out the test issue. 
Changes LGTM

**Comment by cevich** [comment]
Thanks @edsantiago

@TomSweeneyRedHat Thanks for the feedback.  Though I'm willing to bet the same changes are also needed in podman, I'll try to take care of them both.

**Comment by cevich** [comment]
okay @TomSweeneyRedHat this one should be good now.  I manually verified the links, but it may not hurt to double-check.

File: .github/workflows/multi-arch-build.yaml [hunk 6]
```diff
@@ -44,103 +52,149 @@ jobs:
...
+          org.opencontainers.image.source=https://github.com/${{ github.repository }}.git
+          org.opencontainers.image.revision=${{ github.sha }}
+          org.opencontainers.image.created=$(date -u --iso-8601=seconds)
+          DELIMITER
+          EOF
 
       # Separate steps to login and push for buildah and containers quay
       # repositories are required, because 2 sets of credentials are used and `docker
-      # login` as well as `podman login` do not support having 2 different
+      # login` as well as `buildah login` do not support having 2 different
```
Comment (TomSweeneyRedHat) [inline .github/workflows/multi-arch-build.yaml:157]: FWIW, we were talking about this in a meeting earlier today, two or more sets of creds is likely to be allowed in the not too distant future.

**Comment by TomSweeneyRedHat** [comment]
nice work @cevich, LGTM
/hold cancel

**Comment by cevich** [comment]
Thanks guys.  Okay, I'll go press the manual-run button to confirm it actually works :wink:

**Comment by cevich** [comment]
https://github.com/containers/buildah/actions/runs/814748726

**Comment by cevich** [comment]
Build for "testing" failed due to misconfigured repository permissions.  I fixed those and started a new build:

https://github.com/containers/buildah/actions/runs/817182981
